app/stores/ada/AdaRedemptionStore.js

```diff
@@ -0,0 +1,310 @@
+      return; // We cannot decrypt it yet!
+    }
+    this._parseCodeFromCertificate();
+  });
+
+  _setPassPhrase = action(({ passPhrase } : { passPhrase: string }) => {
+    this.passPhrase = passPhrase;
+    if (this.isValidRedemptionMnemonic(passPhrase)) this._parseCodeFromCertificate();
+  });
+
+  _setRedemptionCode = action(({ redemptionCode } : { redemptionCode: string }) => {
+    this.redemptionCode = redemptionCode;
+  });
+
+  _setEmail = action(({ email } : { email: string }) => {
+    this.email = email;
+    this._parseCodeFromCertificate();
+  });
+
+  _setAdaPasscode = action(({ adaPasscode } : { adaPasscode: string }) => {
+    this.adaPasscode = adaPasscode;
+    this._parseCodeFromCertificate();
+  });
+
+  _setAdaAmount = action(({ adaAmount } : { adaAmount: string }) => {
+    this.adaAmount = adaAmount;
+    this._parseCodeFromCertificate();
+  });
+
+  _setDecryptionKey = action(({ decryptionKey } : { decryptionKey: string }) => {
+    this.decryptionKey = decryptionKey;
+    this._parseCodeFromCertificate();
+  });
+
+  _parseCodeFromCertificate() {
+    if (
+      this.redemptionType === ADA_REDEMPTION_TYPES.REGULAR ||
+      this.redemptionType === ADA_REDEMPTION_TYPES.RECOVERY_REGULAR
+    ) {
+      if (!this.passPhrase && this.isCertificateEncrypted) return;
+    }
+    if (this.redemptionType === ADA_REDEMPTION_TYPES.FORCE_VENDED) {
+      if ((!this.email || !this.adaAmount || !this.adaPasscode) && this.isCertificateEncrypted) {
+        return;
+      }
+    }
+    if (this.redemptionType === ADA_REDEMPTION_TYPES.RECOVERY_FORCE_VENDED) {
+      if (!this.decryptionKey && this.isCertificateEncrypted) return;
+    }
+    if (this.redemptionType === ADA_REDEMPTION_TYPES.PAPER_VENDED) return;
+    if (this.certificate === null) throw new NoCertificateError();
+    let decryptionKey = null;
+    if ((
+      this.redemptionType === ADA_REDEMPTION_TYPES.REGULAR ||
+      this.redemptionType === ADA_REDEMPTION_TYPES.RECOVERY_REGULAR) &&
+      this.isCertificateEncrypted
+    ) {
+      decryptionKey = this.passPhrase;
+    }
+    if (
+      this.redemptionType === ADA_REDEMPTION_TYPES.FORCE_VENDED &&
+      this.isCertificateEncrypted
+    ) {
+      decryptionKey = [this.email, this.adaPasscode, this.adaAmount].toString();
+    }
+    if (
+      this.redemptionType === ADA_REDEMPTION_TYPES.RECOVERY_FORCE_VENDED &&
+      this.isCertificateEncrypted
+    ) {
+      decryptionKey = this.decryptionKey;
+    }
+    this.api.ada.getPDFSecretKey(this.certificate, decryptionKey, this.redemptionType)
+      .then(code => this._onCodeParsed(code))
+      .catch(error => this._onParseError(error));
+  }
+
+  _onCodeParsed = action(code => {
+    Logger.debug('Redemption code parsed from certificate: ' + code);
+    this.redemptionCode = code;
+  });
+
+  _onParseError = action(error => {
+    const errorMessage = isString(error) ? error : error.message;
+    if (errorMessage.includes('Invalid mnemonic')) {
```

Comment:
Will this still work on non-English lang? Better to check the `error.id` instead

Reply:
This error doesn't have a translation, but we have to remove its handling anyway because it was thrown by daedalus endpoint, and in yoroi's code we check that those mnemonics are valid in the form.